lib/vsc/filesystem/quota/ktools.py

```diff
@@ -0,0 +1,349 @@
+#
+# Copyright 2020-2020 Ghent University
+#
+# This file is part of vsc-filesystems-quota,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.ugent.be/hpcugent/vsc-filesystems-quota
+#
+# All rights reserved.
+#
+"""
+Tools to sync quota from kafka to the AP
+
+Data was pushed to kafka through gpfsbeat. This is then fetched from kafka and
+pushed to the AP.
+
+@author: Andy Georges (Ghent University)
+"""
+
+import glob
+import logging
+import os
+import re
+
+from collections import defaultdict, namedtuple
+from datetime import datetime
+from kafka import KafkaProducer, KafkaConsumer
+
+from vsc.accountpage.client import AccountpageClient
+from vsc.config.base import GENT_PRODUCTION_COMPUTE_CLUSTERS, STORAGE_SHARED_SUFFIX, VSC
+from vsc.config.base import VSC_SCRATCH_PHANPY, VSC_SCRATCH_ARCANINE, VSC_SCRATCH_KYUKON, VSC_DATA, VSC_HOME
+from vsc.config.base import VO_PREFIX_BY_SITE, VO_SHARED_PREFIX_BY_SITE
+from vsc.config.base import VscStorage, GENT
+from vsc.utils.run import asyncloop, async_run
+from vsc.utils.script_tools import NrpeCLI
+
+from enu34 import Enum
+import json
+
+GPFS_GRACE_REGEX = re.compile(
+    r"(?P<days>\d+)\s*days?|(?P<hours>\d+)\s*hours?|(?P<minutes>\d+)\s*minutes?|(?P<expired>expired)"
+)
+
+GPFS_NOGRACE_REGEX = re.compile(r"none", re.I)
+
+class UsageType(Enum):
+    UserUsage = "USR"
+    FilesetUsage = "FILESET"
+
+
+UsageInformation = namedtuple('UsageInformation', [
+    'used',  # used quota in KiB
+    'soft',  # soft quota limit in KiB
+    'hard',  # hard quota limit in KiB
+    'doubt',  # the KiB GPFS is not sure about
+    'expired',  # tuple (boolean, grace period expressed in seconds)
+    'files_used',  # used number of inodes
+    'files_soft',  # soft limit for inodes
+    'files_hard',  # hard limit for inodes
+    'files_doubt',  # the inodes GPFS is not sure about
+    'files_expired',  # tuple (boolean, grace period expressed in seconds)
+])
+
+# FIXME: pick a better place to do this translation
+fsmap = {
```

Comment:
`/etc/filesystem_info.conf` has this info?

Reply:
Ah, right :)